plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,195 @@
+        }
+        return newValue;
+    };
+}
+
+Blockly.Extensions.registerMutator('procedure_dependencies_mutator', {
+    // Store number of inputs in XML as '<mutation inputs="inputCount_"></mutation>'
+    mutationToDom: function () {
+        const container = document.createElement('mutation');
+        container.setAttribute('inputs', this.inputCount_);
+        return container;
+    },
+
+    // Retrieve number of inputs from XML
+    domToMutation: function (xmlElement) {
+        this.inputCount_ = parseInt(xmlElement.getAttribute('inputs'), 10);
+        this.updateShape_();
+    },
+
+    // Store number of inputs in JSON
+    saveExtraState: function () {
+        return {
+            'inputCount': this.inputCount_
+        };
+    },
+
+    // Retrieve number of inputs from JSON
+    loadExtraState: function (state) {
+        this.inputCount_ = state['inputCount'];
+        this.updateShape_();
+    },
+
+    // "Split" this block into the correct number of inputs in the mutator UI
+    decompose: function (workspace) {
+        const containerBlock = workspace.newBlock('procedure_dependencies_mutator_container');
+        containerBlock.initSvg();
+        let connection = containerBlock.getInput('STACK').connection;
+        for (let i = 0; i < this.inputCount_; i++) {
+            const inputBlock = workspace.newBlock('procedure_dependencies_mutator_input');
+            inputBlock.nameValue_ = null;
+            inputBlock.initSvg();
+            connection.connect(inputBlock.previousConnection);
+            connection = inputBlock.nextConnection;
+        }
+        return containerBlock;
+    },
+
+    // Rebuild this block based on the number of inputs in the mutator UI
+    compose: function (containerBlock) {
+        let inputBlock = containerBlock.getInputTargetBlock('STACK');
+        const connections = [];
+        const fieldValues = {};
+        const fieldValuesDummy = []; // Names of dependency inputs (field values) that don't have any block attached
+        while (inputBlock && !inputBlock.isInsertionMarker()) {
+            connections.push(inputBlock.valueConnection_);
+            if (inputBlock.valueConnection_)
+                fieldValues[inputBlock.valueConnection_.sourceBlock_.id] = inputBlock.nameValue_;
+            else
+                fieldValuesDummy.push(inputBlock.nameValue_);
+            inputBlock = inputBlock.nextConnection && inputBlock.nextConnection.targetBlock();
+        }
+        // Disconnect any children that don't belong
+        for (let i = 0; i < this.inputCount_; i++) {
+            const connection = this.getInput('arg' + i) && this.getInput('arg' + i).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1)
+                connection.disconnect();
+        }
+        this.inputCount_ = connections.length;
+        this.updateShape_();
+        const fieldValuesFlat = Object.values(fieldValues).concat(fieldValuesDummy);

```

Comment:
Some minor notes for readability sake: 
- I'd move this declaration right after the `// Find first index...` comment (as this const is used only there)
- `j` and `k` could be renamed to something more explicit (`dummyIndex` and `defaultNameIndex` maybe?)

Reply:
`j` is used as a counter to iterate `fieldValuesDummy` array though.

Reply:
:heavy_check_mark: Fixed